level2/restaurants/kubo_restaurants.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Restaurant class
+class Restaurant
+  attr_reader :name, :items
+
+  def initialize(name)
+    @name = name
+    @items = []
+  end
+
+  def add_items(*items)
+    items.each { |item| @items << item }
+  end
+
+  def remove_item(item_name)
+    @items.reject! { |item| item.name == item_name }
+  end
+
+  def place_order(order_items, order_discount = 0)
+    order_discount = order_discount[:discount] if order_discount != 0
+    order_price = 0
+    order_items.each do |order_item|
+      checked_item = @items.select { |i| i.name == order_item[:name] }[0]
+      return DOUBLE_DISCOUNT_ERROR if order_item[:discount] && !order_discount.zero?
+      return UNAVAILABLE_ITEM_ERROR unless checked_item
+      return UNAVAILABLE_ITEM_ERROR unless stock_check(order_item, checked_item)
```

Comment:
while this works okay, wouldnt it make more sense if the item was able to tell whether its in stock or not?

Reply:
not sure what you mean, 
`return UNAVAILABLE_ITEM_ERROR unless checked_item` 
this check if item exists in the restaurant items
and
`return UNAVAILABLE_ITEM_ERROR unless stock_check(order_item, checked_item)`
this check if item is in stock

Reply:
I merged them into one as they are returning the same error, but still dont understand what you meant :(

Reply:
sorry, i meant that checking whether item is in stock or not could be responsibility of the `Item` class, so you could just do `item.in_stock?(count)`